File: BookStoreApi/BookStoreApi/Dtos/ProductFullDto.cs
namespace BookStoreApi.Dtos;

/// <summary>
/// Dto товара со всей информацией
/// </summary>
public class ProductFullDto
{
    /// <summary>
    /// Идентификатор
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// Наименование
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// Ссылки на картинки товара
    /// </summary>
    public IEnumerable<string> ImageUrls { get; set; }

    /// <summary>
    /// Цена
    /// </summary>
    public double Price { get; set; }
}

File: frontend/src/app/dtos/product-full.dto.ts
/** Dto товара со всей информацией. */
export class ProductFullDto {
    /** Идентификатор. */
    public id: number;

    /** Наименование. */
    public name: string;

    /** Гуиды картинок товара. */
    public imageGuids: string[];

    /** Цена. */
    public price: number;

    constructor(id: number, name: string, imageGuids: string[], price: number) {
        this.id = id;
        this.name = name;
        this.imageGuids = imageGuids;
        this.price = price;
    }
}
